program_transformers/src/batch_minting/batch_mint_persister.rs

```diff
@@ -465,30 +347,49 @@ impl<T: ConnectionTrait + TransactionTrait, D: BatchMintDownloader> BatchMintPer
         batch_mint_to_verify: &mut batch_mint_to_verify::Model,
         batch_mint: &BatchMint,
     ) {
+        let collection_raw_key: Option<[u8; 32]> =
+            if let Some(key) = batch_mint_to_verify.collection.clone() {
```

Comment:
maybe use map()?

Reply:
the trick is that we need to return if key convertion was failed and we cannot do it from map so alternative code with map using would look like this:
```
let collection_raw_key: Option<[u8; 32]> = batch_mint_to_verify.collection.clone().map(|key| {
    match key.try_into() {
        Ok(key_as_array) => Some(key_as_array),
        Err(e) => {
            error!("Could not convert collection key received from DB: {:?}", e);

            self.mark_persisting_failed(batch_mint_to_verify).await;
            None
        }
    }
}).flatten();

if collection_raw_key.is_none() && batch_mint_to_verify.collection.is_some() {
    return;
}
```

in my opinion this refactor will not really improve code readability.